File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for(int i=0;i<nums.length-1;i++){
            int a = nums[i];
            for(int j = i+1;j<nums.length;j++){
                if(nums[j]==a)
                return true;
            }
            
        }
        return false;
    }
}

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        cache = [-1]*n

        def dfs(i):
            if i<0 or i>=n:
                return 0
            if cache[i] != -1:
                return cache[i]
            cache[i] = max(nums[i]+dfs(i+2),dfs(i+1))
            return cache[i]
        
        return dfs(0)

